provider/kubernetes/kubernetes.go

```diff
@@ -367,23 +372,35 @@ func (p *Provider) loadConfig(templateObjects types.Configuration) *types.Config
 	return configuration
 }
 
-func getRuleForPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress) string {
+func getRuleForPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress) (string, error) {
 	if len(pa.Path) == 0 {
-		return ""
+		return "", nil
 	}
 
 	ruleType := i.Annotations[label.TraefikFrontendRuleType]
 	if ruleType == "" {
 		ruleType = ruleTypePathPrefix
 	}
 
+	pathReplaced := false
+	if ruleType == ruleTypeReplacePath {
+		pathReplaced = true
+	}
+
 	rules := []string{ruleType + ":" + pa.Path}
 
 	if rewriteTarget := i.Annotations[annotationKubernetesRewriteTarget]; rewriteTarget != "" {
 		rules = append(rules, ruleTypeReplacePath+":"+rewriteTarget)
+		pathReplaced = true
 	}
 
-	return strings.Join(rules, ";")
+	if rootPath := label.GetStringValue(i.Annotations, annotationKubernetesAppRoot, ""); rootPath != "" && pa.Path == "/" {
+		if pathReplaced {
+			return "", fmt.Errorf("app-root shouldn't be use together with other rules of replacepath type")
```

Comment:
Can we phrase this more strongly by replacing _shouldn't_ with _must not_? The former gives some kind of indication that using multiple rules may still be acceptable, which it isn't.